### PR TITLE
remove detectmissingcss method

### PR DESCRIPTION
**Pull request by dbauszus-glx**
Remove missingcss method and mapboxgl-canary dom element. The only purpose is checking whether the css is added and warn if not. This is unnecessary bloat.

#74 (comment)

## Launch Checklist

 - [ ] confirm your changes do not include backports from Mapbox projects (unless with compliant license)
 - [ ] briefly describe the changes in this PR
 - [ ] include before/after visuals or gifs if this PR includes visual changes
 - [ ] write tests for all new functionality
 - [ ] document any changes to public APIs
 - [ ] post benchmark scores
 - [ ] manually test the debug page
 - [ ] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [ ] add an entry inside this element for inclusion in the `maplibre-gl-js` changelog: `<changelog></changelog>`

### CODE REVIEW

**Comment by HarelM** [comment]
Thanks for the PR! please address failing CI runs...

**Comment by dbauszus-glx** [comment]
I did a commit to address the failing CI runs. Not sure how to re-run the tests. Should this be automatic?

**Comment by HarelM** [comment]
It should, not sure why I need to approve the run again... Let me know if this happens again...

**Comment by dbauszus-glx** [comment]
Ah yes, seems to be automatic. Checks running now.

**Comment by dbauszus-glx** [comment]
There was one outstanding issue which should have been addressed with the last commit. The childnode equality should be on 2 not 3 without the mapboxgl-canary element.

**Comment by dbauszus-glx** [comment]
All checks have passed now.

File: test/unit/ui/handler/box_zoom.test.js [hunk 1]
```diff
@@ -5,7 +5,6 @@ import DOM from '../../../../src/util/dom';
 import simulate from '../../../util/simulate_interaction';
 
 function createMap(t, clickTolerance) {
-    t.stub(Map.prototype, '_detectMissingCSS');
```
Comment (HarelM) [inline test/unit/ui/handler/box_zoom.test.js:8]: is `t` still in use here? if not, let's remove it as well like other cases...

File: test/unit/ui/handler/drag_pan.test.js [hunk 2]
```diff
@@ -5,7 +5,6 @@ import DOM from '../../../../src/util/dom';
 import simulate from '../../../util/simulate_interaction';
 
 function createMap(t, clickTolerance, dragPan) {
-    t.stub(Map.prototype, '_detectMissingCSS');
```
Comment (HarelM) [inline test/unit/ui/handler/drag_pan.test.js:8]: Same as above

File: test/unit/ui/handler/drag_rotate.test.js [hunk 3]
```diff
@@ -7,7 +7,6 @@ import simulate from '../../../util/simulate_interaction';
 import browser from '../../../../src/util/browser';
 
 function createMap(t, options) {
-    t.stub(Map.prototype, '_detectMissingCSS');
```
Comment (HarelM) [inline test/unit/ui/handler/drag_rotate.test.js:10]: Same here

File: test/unit/ui/handler/keyboard.test.js [hunk 4]
```diff
@@ -6,7 +6,6 @@ import simulate from '../../../util/simulate_interaction';
 import {extend} from '../../../../src/util/util';
 
 function createMap(t, options) {
-    t.stub(Map.prototype, '_detectMissingCSS');
```
Comment (HarelM) [inline test/unit/ui/handler/keyboard.test.js:9]: Same here

File: test/unit/ui/handler/map_event.test.js [hunk 5]
```diff
@@ -4,8 +4,7 @@ import Map from '../../../../src/ui/map';
 import DOM from '../../../../src/util/dom';
 import simulate from '../../../util/simulate_interaction';
 
-function createMap(t) {
-    t.stub(Map.prototype, '_detectMissingCSS');
+function createMap() {
```
Comment (HarelM) [inline test/unit/ui/handler/map_event.test.js:7]: Good! :-)

File: test/unit/ui/handler/mouse_rotate.test.js [hunk 6]
```diff
@@ -7,7 +7,6 @@ import simulate from '../../../util/simulate_interaction';
 import browser from '../../../../src/util/browser';
 
 function createMap(t, options) {
-    t.stub(Map.prototype, '_detectMissingCSS');
```
Comment (HarelM) [inline test/unit/ui/handler/mouse_rotate.test.js:10]: Same

File: test/util/index.js [hunk 7]
```diff
@@ -19,7 +19,6 @@ export function createMap(t, options, callback) {
     Object.defineProperty(container, 'clientWidth', {value: 200, configurable: true});
     Object.defineProperty(container, 'clientHeight', {value: 200, configurable: true});
 
-    if (!options || !options.skipCSSStub) t.stub(Map.prototype, '_detectMissingCSS');
```
Comment (HarelM) [inline test/util/index.js:22]: Is `skipCSSStub` used elsewhere? probably a good idea to remove this too if it is no longer in use.

**Review by HarelM — CHANGES_REQUESTED** [review]
Small stuff, thanks.

File: test/util/index.js (hunk 7 above)
Comment (dbauszus-glx) [reply test/util/index.js:22]: I removed skipcssstub. https://github.com/maplibre/maplibre-gl-js/pull/193/commits/71174418354ae5e134a68dc5838418bb5969e99d

File: test/unit/ui/handler/box_zoom.test.js (hunk 1 above)
Comment (dbauszus-glx) [reply test/unit/ui/handler/box_zoom.test.js:8]: should be resolved here https://github.com/maplibre/maplibre-gl-js/pull/193/commits/42b047d3674503ab2b3091c397d48ba17df05332

File: test/unit/ui/handler/drag_pan.test.js (hunk 2 above)
Comment (dbauszus-glx) [reply test/unit/ui/handler/drag_pan.test.js:8]: should be resolved here https://github.com/maplibre/maplibre-gl-js/pull/193/commits/42b047d3674503ab2b3091c397d48ba17df05332

File: test/unit/ui/handler/drag_rotate.test.js (hunk 3 above)
Comment (dbauszus-glx) [reply test/unit/ui/handler/drag_rotate.test.js:10]: should be resolved here https://github.com/maplibre/maplibre-gl-js/pull/193/commits/42b047d3674503ab2b3091c397d48ba17df05332

File: test/unit/ui/handler/keyboard.test.js (hunk 4 above)
Comment (dbauszus-glx) [reply test/unit/ui/handler/keyboard.test.js:9]: should be resolved here https://github.com/maplibre/maplibre-gl-js/pull/193/commits/42b047d3674503ab2b3091c397d48ba17df05332

File: test/unit/ui/handler/mouse_rotate.test.js (hunk 6 above)
Comment (dbauszus-glx) [reply test/unit/ui/handler/mouse_rotate.test.js:10]: should be resolved here https://github.com/maplibre/maplibre-gl-js/pull/193/commits/42b047d3674503ab2b3091c397d48ba17df05332

**Comment by dbauszus-glx** [comment]
The restest for macos failed due to trouble with the network connection.

**Comment by dbauszus-glx** [comment]
All the requested changes should now be resolved.

File: test/unit/ui/handler/drag_pan.test.js (hunk 2 above)
Comment (dbauszus-glx) [reply test/unit/ui/handler/drag_pan.test.js:8]: I don't think we can remove the first argument.

This causes tests to fail which access the first argument. t.equal(map.easeTo.getCall(0).args[0].zoom, 11, 'pressing the +/= key should
      zoom map in');

Comment (HarelM) [reply test/unit/ui/handler/drag_pan.test.js:8]: I think you changed the method signature but didn't change the usage in the tests that use it...? I might be wrong...

**Comment by HarelM** [comment]
I think that the removal of the `t` was OK, but the usage of the method was not altered and that what I think caused the test to fail.
Can you try and bring this back and fix the tests method usage as well.
I'm not sure the last failed test is not a flickering test, but I would like to rerun them after the removal of the `t` if this is OK by you...

**Comment by dbauszus-glx** [comment]
I took them out again and will see what the checks come back with.

**Comment by HarelM** [comment]
That won't work man... your removing the t but don't change the method call...
I'll add a comment for example...

File: test/unit/ui/handler/box_zoom.test.js [hunk 8]
```diff
@@ -4,7 +4,7 @@ import Map from '../../../../src/ui/map';
 import DOM from '../../../../src/util/dom';
 import simulate from '../../../util/simulate_interaction';
 
-function createMap(t, clickTolerance) {
+function createMap(clickTolerance) {
```
Comment (HarelM) [inline test/unit/ui/handler/box_zoom.test.js:7]: In line 12 here you need to remove the `t` when you call this method...

File: test/unit/ui/handler/drag_pan.test.js [hunk 9]
```diff
@@ -4,7 +4,7 @@ import Map from '../../../../src/ui/map';
 import DOM from '../../../../src/util/dom';
 import simulate from '../../../util/simulate_interaction';
 
-function createMap(t, clickTolerance, dragPan) {
+function createMap(clickTolerance, dragPan) {
```
Comment (HarelM) [inline test/unit/ui/handler/drag_pan.test.js:7]: Same here, on line 19...

File: test/unit/ui/handler/drag_rotate.test.js [hunk 10]
```diff
@@ -6,7 +6,7 @@ import DOM from '../../../../src/util/dom';
 import simulate from '../../../util/simulate_interaction';
 import browser from '../../../../src/util/browser';
 
-function createMap(t, options) {
+function createMap(options) {
```
Comment (HarelM) [inline test/unit/ui/handler/drag_rotate.test.js:9]: Same here, in line 13 and probably in other lines...

File: test/unit/ui/handler/box_zoom.test.js (hunk 8 above)
Comment (dbauszus-glx) [reply test/unit/ui/handler/box_zoom.test.js:7]: Thanks for the comment. I didn't get this at first. Have changed these function calls now.

**Comment by dbauszus-glx** [comment]
should be good now.

**Review by HarelM — APPROVED** [review]

**Comment by HarelM** [comment]
@wipfli @nyurik let me know if you would like to review this.
If not, I'll simply merge this.

**Pull request merged**